blaze-client/src/main/scala/org/http4s/blaze/client/Connection.scala

```diff
@@ -26,7 +26,7 @@ private[client] trait Connection[F[_]] {
   def isClosed: Boolean
 
   /** Determine if the connection is in a state that it can be recycled for another request. */
-  def isRecyclable: Boolean
+  def isRecyclable: F[Boolean]
```

Comment:
Some other methods here could get the same treatment, but I want to get this landed.